test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
-		);
+		it('should call library.logger.trace with "Blocks->Utils: Submodule initialized."', () => {
+			expect(loggerStub.trace.args[0][0]).to.equal(
+				'Blocks->Utils: Submodule initialized.'
+			);
+		});
 
-		it('should return self');
+		it('should return self', () => {
+			expect(blocksUtilsModule).to.be.an('object');
+			expect(blocksUtilsModule.readDbRows).to.be.a('function');
+		});
 	});
 
 	describe('readDbRows', () => {
-		it('should loop through all given rows');
-
-		it('should call library.logic.block.dbRead with every row');
-
-		describe('when block with id does not exist', () => {
-			it('should not return the block');
+		it('should transform a full_blocks_list view row into a block object', () => {
+			var blockObject = blocksUtilsModule.readDbRows(viewRow_full_blocks_list);
+
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('13068833527549895884');
+			expect(blockObject[0].height).to.equal(3);
+			expect(blockObject[0].transactions).to.be.an('array');
+			expect(blockObject[0].transactions[0]).to.be.an('object');
+			expect(blockObject[0].transactions[0].id).to.equal('6950874693022090568');
+			expect(blockObject[0].transactions[0].type).to.equal(0);
 		});
 
-		describe('when block with id exists', () => {
-			describe('and block indices are duplicated', () => {
-				it('should not return duplicated blocks');
-			});
-
-			describe('and block indices are unique', () => {
-				describe('and there are no transactions in the block', () => {
-					it('should return the block containing transactions = {}');
-				});
+		it('should generate fake signature for genesis block', () => {
+			var genesisBlock_view_full_blocks_list = [
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '1465651642158264047',
+					t_type: 0,
+				},
+				{
+					b_id: '6524861224470851795',
+					b_height: 1,
+					t_id: '3634383815892709956',
+					t_type: 2,
+				},
+			];
+
+			var blockObject = blocksUtilsModule.readDbRows(
+				genesisBlock_view_full_blocks_list
+			);
 
-				describe('and there are transactions in the block', () => {
-					it('should return the block containing transactions');
-				});
-			});
+			expect(blockObject).to.be.an('array');
+			expect(blockObject[0]).to.be.an('object');
+			expect(blockObject[0].id).to.equal('6524861224470851795');
+			expect(blockObject[0].generationSignature).to.equal(
+				'0000000000000000000000000000000000000000000000000000000000000000'
+			);
 		});
 	});
 
 	describe('loadBlocksPart', () => {
-		describe('when there is no error when loading block array', () => {
-			it('should be normalized');
-
-			it('should call callback with error = undefined');
-
-			it('should call callback with blocks as result');
+		it('should return error when library.db.blocks.loadBlocksData fails', done => {
+			library.db.blocks.getHeightByLastId = sinonSandbox.stub().resolves([]);
```

Comment:
why we stub that function here?

Reply:
Because we are testing the child promise errors are caught.

Reply:
`loadBlocksPart` is calling `loadBlocksData` that we stub, so I don't see a point of stubbing `library.db.blocks.getHeightByLastId` here, as it's not called at all.

Reply:
Yes ok, will remove.

Reply:
Done.